File: dotnet/picocontainer/PicoContainer.Source/PicoContainer.Defaults/NullComponentMonitor.cs
using System;
using System.Reflection;

namespace PicoContainer.Core.Defaults
{
	[Serializable]
	public class NullComponentMonitor : IComponentMonitor
	{
		public void Instantiating(ConstructorInfo constructor)
		{
		}

		public void Instantiated(ConstructorInfo constructor, long beforeTime, long duration)
		{
		}

		public void InstantiationFailed(ConstructorInfo constructor, Exception e)
		{
		}

		public void Invoking(MethodInfo method, Object instance)
		{
		}

		public void Invoked(MethodInfo method, Object instance, long duration)
		{
		}

		public void InvocationFailed(MethodInfo method, Object instance, Exception e)
		{
		}
	}
}

File: java/picocontainer/tags/picocontainer-1_3/container/src/java/org/picocontainer/monitors/NullComponentMonitor.java
/*****************************************************************************
 * Copyright (C) PicoContainer Organization. All rights reserved.            *
 * ------------------------------------------------------------------------- *
 * The software in this package is published under the terms of the BSD      *
 * style license a copy of which has been included with this distribution in *
 * the LICENSE.txt file.                                                     *
 *                                                                           *
 * Original code by Paul Hammant & Obie Fernandez & Aslak Helles&oslash;y    *
 *****************************************************************************/

package org.picocontainer.monitors;

import java.io.Serializable;
import java.lang.reflect.Constructor;
import java.lang.reflect.Method;

import org.picocontainer.ComponentMonitor;

/**
 * A {@link ComponentMonitor} which does nothing. 
 * 
 * @author Paul Hammant
 * @author Obie Fernandez
 * @version $Revision$
 */
public class NullComponentMonitor implements ComponentMonitor, Serializable {

    private static NullComponentMonitor instance;

    public void instantiating(Constructor constructor) {
    }

    public void instantiated(Constructor constructor, long duration) {
    }

    public void instantiationFailed(Constructor constructor, Exception e) {
    }

    public void instantiated(Constructor constructor, Object instantiated, Object[] injected, long duration) {
    }

    public void invoking(Method method, Object instance) {
    }

    public void invoked(Method method, Object instance, long duration) {
    }

    public void invocationFailed(Method method, Object instance, Exception e) {
    }

    public void lifecycleInvocationFailed(Method method, Object instance, RuntimeException cause) {
    }

    public static synchronized NullComponentMonitor getInstance() {
        if (instance == null) {
            instance = new NullComponentMonitor();
        }
        return instance;
    }
}
